share_app/src/main/java/org/odk/share/adapters/InstanceAdapter.java

```diff
@@ -0,0 +1,85 @@
+package org.odk.share.adapters;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.TextView;
+
+import org.odk.share.R;
+import org.odk.share.provider.InstanceProviderAPI;
+
+import java.util.LinkedHashSet;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * Created by laksh on 5/20/2018.
+ */
+
+public class InstanceAdapter extends RecyclerView.Adapter<InstanceAdapter.InstanceHolder> {
+
+    private Cursor cursor;
+    private Context context;
+    private final OnItemClickListener listener;
+    private LinkedHashSet<Long> selectedInstances;
+
+    public InstanceAdapter(Context context, Cursor cursor, OnItemClickListener listener,
+                           LinkedHashSet<Long> selectedInstances) {
+        this.context = context;
+        this.cursor = cursor;
+        this.listener = listener;
+        this.selectedInstances = selectedInstances;
+    }
+
+    @Override
+    public InstanceHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.list_item_checkbox, null);
+        return new InstanceHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull InstanceHolder holder, int position) {
+        cursor.moveToPosition(holder.getAdapterPosition());
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                listener.onItemClick(v, holder.getAdapterPosition());
+            }
+        });
+        holder.title.setText(cursor.getString(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns.DISPLAY_NAME)));
+        holder.subtitle.setText(cursor.getString(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns.DISPLAY_SUBTEXT)));
+        long id = cursor.getLong(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns._ID));
+        holder.checkBox.setChecked(selectedInstances.contains(id));
+    }
+
+    @Override
+    public int getItemCount() {
+        return cursor.getCount();
+    }
+
+    public class InstanceHolder extends RecyclerView.ViewHolder {
```

Comment:
Class declaration should be at the bottom. Just try  IntelliJ's auto-reformat `Ctrl` + 'Shift` + 'f'

Reply:
Also, the scope should be package-private as this holder is not going to be used outside of this class